pkg/operator/ceph/object/controller.go

```diff
@@ -519,3 +522,48 @@ func (r *ReconcileCephObjectStore) reconcileMultisiteCRs(cephObjectStore *cephv1
 
 	return cephObjectStore.Name, cephObjectStore.Name, cephObjectStore.Name, nil, reconcile.Result{}, nil
 }
+
+func (r *ReconcileCephObjectStore) reconcileCOSIUser(cephObjectStore *cephv1.CephObjectStore) (reconcile.Result, error) {
+	// Create COSI user and secret
+	userConfig := generateCOSIUserConfig()
+
+	// Create COSI user
+	objCtx, err := NewMultisiteContext(r.context, r.clusterInfo, cephObjectStore)
+	if err != nil {
+		return reconcile.Result{}, errors.Wrapf(err, "failed to get object context")
+	}
+
+	adminOpsCtx, err := NewMultisiteAdminOpsContext(objCtx, &cephObjectStore.Spec)
+	if err != nil {
+		return reconcile.Result{}, errors.Wrapf(err, "failed to get admin ops API context")
+	}
+
+	_, err = adminOpsCtx.AdminOpsClient.GetUser(r.opManagerContext, *userConfig)
+	if err != nil {
+		if errors.Is(err, admin.ErrNoSuchUser) {
+			logger.Infof("creating COSI user %q", userConfig.ID)
+			_, err = adminOpsCtx.AdminOpsClient.CreateUser(r.opManagerContext, *userConfig)
+			if err != nil {
+				return reconcile.Result{}, errors.Wrapf(err, "failed to create COSI user %q", userConfig.ID)
+			}
+		} else {
+			return reconcile.Result{}, errors.Wrapf(err, "failed to get COSI user %q", userConfig.ID)
+		}
+	}
+
+	secretName := fmt.Sprintf("%s-%s", cephObjectStore.Name, cosiUserName)
+	// Create COSI user secret
+	return ReconcileCephUserSecret(r.opManagerContext, r.client, r.scheme, cephObjectStore, userConfig, objCtx.Endpoint, secretName, cephObjectStore.Namespace, cephObjectStore.Name, cephObjectStore.Spec.Gateway.SSLCertificateRef)
+}
+
+func generateCOSIUserConfig() *admin.User {
```

Comment:
Is this method not used?

Reply:
L:528, In the beginning of reconcileCOSIUser